.github/workflows/rust.yml

```diff
@@ -38,6 +38,11 @@ jobs:
     steps:
     - uses: actions/checkout@v3
 
+    - name: Setup end2end dependencies
+      run: |
+        docker compose up -d --wait
```

Comment:
Could we use [service containers](https://docs.github.com/en/actions/using-containerized-services/about-service-containers) instead? I appreciate the reuse here but honestly I haven't ever seen this used in CI in a way that isn't flakey.

Reply:
I'll pull a "trust me" on this one if you don't mind. Happy to switch to something else if we do observe flakiness.
I hate the commit-and-push-and-wait-for-action-to-run feedback loop, so having a compose I can spin up locally makes iterating and debugging faster.

Reply:
Sure. I will say that I absolutely despise docker compose both in CI and locally, but I'll workaround for now. Generally services are cached and much faster